lib/sync/fullsync.ts

```diff
@@ -10,11 +11,24 @@ import { BlockFetcher } from './fetcher'
  * @memberof module:sync
  */
 export class FullSynchronizer extends Synchronizer {
+  public vm: VM
+
   private blockFetcher: BlockFetcher | null
 
   constructor(options: SynchronizerOptions) {
     super(options)
     this.blockFetcher = null
+
+    if (!this.config.vm) {
+      this.vm = new VM({
+        common: this.config.common,
+        blockchain: this.chain.blockchain,
+      })
+    } else {
+      this.vm = this.config.vm
+      //@ts-ignore blockchain has readonly property
+      this.vm.blockchain = this.chain.blockchain
```

Comment:
CodeCov reports that this is not covered by tests, could you add a test that ensures that the configured VM is indeed the correct one?